src/profile-logic/symbolication.js

```diff
@@ -98,30 +127,31 @@ function gatherFuncsInThread(thread) {
  * Using the provided func address table, find out which funcs are actually the
  * same function, and make a list of the real func addresses that we need
  * symbols for.
+ *
  * Before we had the func address table for this library, we weren't able to
  * tell whether two frames are the same function, so we naively created one
  * function per frame, giving each function the address of the frame.
  * Now that we know at which address each function truly starts, we can find
  * out which of the naively-created funcs are the same function and collapse
  * those into just one func. This information is returned in the
- * oldFuncToNewFuncMap outparameter.
+ * oldFuncToNewFuncMap.
+ *
  * Before we can request symbols, we need to make a list of func addresses for
  * the symbols we need. These addresses need to be the true function start
  * addresses from the func address table. This information is returned in the
  * return value - the return value is a map whose keys are the true func
  * addresses that we need symbols for. The value of each map entry is the
  * funcIndex for the func that has become the "one true func" for this function.
- * @param  {Array}  funcAddressTable    An array containing the address of every function of a library, in ascending order.
- * @param  {Array}  funcsToSymbolicate  An array containing funcIndex elements for the funcs in this library.
- * @param  {Object} funcTable           The funcTable that the funcIndices in addressToSymbolicate refer to.
- * @param  {Map}    oldFuncToNewFuncMap An out parameter that specifies how funcs should be merged.
- * @return {Object}                     A map that maps a func address index to a funcIndex, one entry for each func that needs to be symbolicated.
  */
 function findFunctionsToMergeAndSymbolicationAddresses(
-  funcAddressTable,
-  funcsToSymbolicate,
-  funcTable
-) {
+  funcAddressTable: Uint32Array,
+  funcsToSymbolicate: IndexIntoFuncTable[],
+  funcTable: FuncTable
+): {|
+  funcAddrIndices: IndexIntoAddressTable[],
+  funcIndices: IndexIntoFuncTable[],
+  oldFuncToNewFuncMap: Map<IndexIntoFuncTable, IndexIntoFuncTable>,
+|} {
```

Comment:
Hmm... the results are just immediately destructured in its one usage, so I think it's not worth giving it a name, even though it is a little complicated..